pkg/component/shared/resourcemanager.go

```diff
@@ -348,3 +291,19 @@ func waitUntilGardenerResourceManagerBootstrapped(ctx context.Context, c client.
 
 	return managedresources.WaitUntilHealthy(ctx, c, namespace, resourcemanager.ManagedResourceName)
 }
+
+func applyDefaults(userValues *resourcemanager.Values, defaultValues resourcemanager.Values) {
+	var (
+		vUser    = reflect.ValueOf(userValues).Elem()
+		vDefault = reflect.ValueOf(defaultValues)
+	)
+
+	for i := 0; i < vUser.NumField(); i++ {
+		userField := vUser.Field(i)
+		defaultField := vDefault.Field(i)
+
+		if reflect.DeepEqual(userField.Interface(), reflect.Zero(userField.Type()).Interface()) {
```

Comment:
I have not used reflection in `go`. Therefore, I might fail to see the obvious here.

Nevertheless, why is `userField.isZero()` not sufficient here? At least from the documentation perspective it looks like it should do the job.